src/aia_clic_extension.adoc

```diff
@@ -0,0 +1,1816 @@
+
+28: xpepc (XXLEN)
+
+** XXLEN = 32
+
+stack pointer adjustment: 64
+
+60: reserved (5) | xpp (2) |  xpie (1) |  xpil (8) | xpinterrupt (1) | reserved (3) | xpexccode (12)
+
+56: xpepc (XXLEN)
+
+=== Fault handling
+
+According the the Machine Level ISA, trap handlers must be designed to not cause exceptions during their critical section.
+This applies to the ipush and ipopxret instruction.
+Therefore, by design, it must be guaranteed that no faults occur during the execution of these instructions.
+
+
+=== Debug
+
+As the execution of ipush may start before the instruction is fetched, breakpoint conditions met might observe an inconsistent state.
+Therefore, if a breakpoint is triggered because of an ipush instruction, the breakpoint shall be handled as if it occured on the subsequent instruction.
+
+=== Instructions
+
+#TODO: define encodings, sail etc.
+
+==== ipush
+
+==== ipopxret
+
+== smcliclat extension
+
+When not in CLIC mode, {mcause} has the CLINT mode format.
+
+
+== CLIC Parameters
+
+Although these are described as parameters, it is understood that hardware implementations may wish to
+have a single implementation support different parameterizations of CLIC extensions and may make
+these values configurable and initialized prior to CLIC operation. 
+However, these parameters should functionally be considered static. If the value of these parameters are changed
+during CLIC operation, CLIC behavior is undefined.
+
+=== NVBITS Parameter - Specifying Support for smclicshv Selective Interrupt Hardware Vectoring Extension
+
+The NVBITS Parameter specifies whether
+the smclicshv extension is implemented or not.
+
+When NVBITS is 0, the smclicshv extension is not implemented.
+In this case, all CLIC interrupts are software vectored and are directed to the common code
+at {tvec} register.
+
+When NVBITS is 1, smclicshv extension is implemented.
+
+=== CLICINFO Parameters
+
+The NUM_INTERRUPT 13-bit parameter from 2-4096 that specifies the actual number of maximum interrupt
+inputs supported in this implementation. MSIP, MTIP are always included.
+
+The VERSION 8-bit parameter specifies the implementation version of CLIC. The upper
+4-bit specifies the architecture version, and the lower 4-bit specifies
+the implementation version.
+
+The NUM_TRIGGER 6-bit parameter specifies the number of maximum interrupt
+triggers supported in this implementation. Valid values are 0 to 32.
+
+=== Additional CLIC Parameters
+
+[source]
+----
+Name           Value Range  Description
+CLICLEVELS     2-256        Number of interrupt levels including 0
+CLICMAXID      12-4095      Largest interrupt ID
+
+INTTHRESHBITS  1-8          Number of bits implemented in {intthresh}.th
+CLICMTVECALIGN >= 2         Number of hardwired-zero LSBs in mtvec address.
+----
+NOTE: These parameters are likely to be available by the general
+discovery mechanism that is in development.
+
+
+== Support for reduced csr accesses in interrupt handlers - smcliclat

```

Comment:
I think it would be good to remove this. We can always bring it back if needed, but this would increase clarity of the current proposal.